### ISSUE TITLE
.dvcignore - new lines throw error, expected str instance, NoneType found.

### ISSUE THREAD

**Issue by melkiga** [iterative/dvc#4064]
```
DVC version: 1.0.0b2
Python version: 3.7.6
Platform: Ubuntu 18.04.4 LTS
Package: dvc_1.0.0b2_amd64.deb
Filesystem: ceph-fuse
```
- **Issue:** After installing DVC v1.0.0b2, all DVC commands fail with the following error:
` ERROR: unexpected error - sequence item 0: expected str instance, NoneType found`
- **Debug:** for command `dvc status -vvv`
```
2020-06-17 21:57:10,649 TRACE: Namespace(all_branches=False, all_commits=False, all_tags=False, cloud=False, cmd='status', func=<class 'dvc.command.status.CmdDataStatus'>, jobs=None, quiet=False, recursive=False, remote=None, show_json=False, targets=[], verbose=3, version=None, with_deps=False)
2020-06-17 21:57:10,904 TRACE: PRAGMA user_version;                             
2020-06-17 21:57:10,905 DEBUG: fetched: [(3,)]
2020-06-17 21:57:10,906 TRACE: CREATE TABLE IF NOT EXISTS state (inode INTEGER PRIMARY KEY, mtime TEXT NOT NULL, size TEXT NOT NULL, md5 TEXT NOT NULL, timestamp TEXT NOT NULL)
2020-06-17 21:57:10,908 TRACE: CREATE TABLE IF NOT EXISTS state_info (count INTEGER)
2020-06-17 21:57:10,910 TRACE: CREATE TABLE IF NOT EXISTS link_state (path TEXT PRIMARY KEY, inode INTEGER NOT NULL, mtime TEXT NOT NULL)
2020-06-17 21:57:10,911 TRACE: INSERT OR IGNORE INTO state_info (count) SELECT 0 WHERE NOT EXISTS (SELECT * FROM state_info)
2020-06-17 21:57:10,913 TRACE: PRAGMA user_version = 3;
2020-06-17 21:57:10,924 TRACE: SELECT count from state_info WHERE rowid=?
2020-06-17 21:57:10,924 DEBUG: fetched: [(0,)]
2020-06-17 21:57:10,924 TRACE: UPDATE state_info SET count = ? WHERE rowid = ?
2020-06-17 21:57:10,987 ERROR: failed to obtain data status - sequence item 0: expected str instance, NoneType found
------------------------------------------------------------
Traceback (most recent call last):
  File "dvc/command/status.py", line 51, in run
  File "dvc/repo/__init__.py", line 35, in wrapper
  File "dvc/repo/status.py", line 146, in status
  File "dvc/repo/status.py", line 35, in _local_status
  File "dvc/repo/__init__.py", line 246, in collect
  File "site-packages/funcy/objects.py", line 28, in __get__
  File "dvc/repo/__init__.py", line 516, in stages
  File "dvc/repo/__init__.py", line 528, in _collect_stages
  File "dvc/ignore.py", line 234, in walk
  File "site-packages/funcy/objects.py", line 28, in __get__
  File "dvc/ignore.py", line 149, in dvcignore
  File "dvc/ignore.py", line 124, in __init__
  File "dvc/ignore.py", line 130, in _update
  File "dvc/ignore.py", line 40, in __init__
  File "dvc/ignore.py", line 39, in <listcomp>
TypeError: sequence item 0: expected str instance, NoneType found
------------------------------------------------------------
```
- **What I did:** I noticed the filename in the trace was `ignore.py`, so I had a hunch it had something to do with the `.dvcignore` file. After I opened mine, I noticed that there were some new-lines between the list of files. After removing all empty lines, the error was resolved. Here is an example `.dvcignore` that would throw the error:
```
raw/tanzil
raw/qed
raw/youtube-bb
raw/vis-drone-2019

raw/wiki
raw/cars-overhead-with-context
raw/unpc
raw/kde
```
- **TL;DR:** New lines in the `.dvcignore` file will cause all DVC commands to fail with the following error `TypeError: sequence item 0: expected str instance, NoneType found`

**Comment by pmrowla** [comment]
Can be reproduced in master with
```py
def test_ignore_newline(tmp_dir, dvc, monkeypatch):
    tmp_dir.gen({"dir": {"ignored": "text", "other": "text2"}})
    tmp_dir.gen(DvcIgnore.DVCIGNORE_FILE, "dir/ignored\n\nfoo")

    assert _files_set("dir", dvc.tree) == {"dir/other"}

    monkeypatch.chdir("dir")
    assert _files_set(".", dvc.tree) == {"./other"}
```
